### DISCUSSION TITLE
Is it possible to rotate link label via UI?

### DISCUSSION THREAD

**Discussion by Lijesh-k** [clientIO/joint#1989] [Q&A]
![211834637-10cdc28f-6a23-4cd8-bac7-3e5119137760](https://user-images.githubusercontent.com/122600888/212840505-467eadb1-d873-4ec7-9372-ceb17a02fe89.png)
I need to rotate the link label vial UI

**Comment by kumilingus** [comment]
It's not supported yet.

If you want to use existing tools like `ui.Halo` or `ui.FreeTransform` you need to switch from labels to JointJS elements. That helps if your plan is also resize the labels.

In order to rotate a label you have to change a single property on the link.
```js
const labelIndex = 1;
const angle = 45;
model.prop(['labels', labelIndex, 'position', 'angle'], angle);
```
You can write a custom `linkTool` that would rotate a specific label.

---

Now this is a little hack because it uses `elementTools` instead of `linkTools`.

https://jsfiddle.net/kumilingus/q3s7doua/

It is a valid use case and we will add support of `Control` tool for links to the library at some point. We also make sure the `selector` option can also target labels and the tool is updated when the labels are moved.

**Reply by Lijesh-k** [reply]
Thanks, @kumilingus 
Is it possible to set an icon or image for the tool?

**Reply by kumilingus** [reply]
It's defined as `<circle/>` ([source code](https://github.com/clientIO/joint/blob/master/src/elementTools/Control.mjs#L7-L15)). You can override it when extending the class. Example [here](https://codepen.io/jointjs/pen/poaEpKx).

**Reply by Lijesh-k** [reply]
Hi @kumilingus 
is there any way to do that with the help of a button
eg: instead of dragging a button when I click the particular label button I need to change the label angle by 90 (add angle by 90)

**Reply by kumilingus** [reply]
For that, use [linkTool.Button](https://resources.jointjs.com/docs/jointjs/v3.6/joint.html#linkTools.Button). Please refer to the documentation and to this thread.

---

Please try to provide as much information as possible the next time you ask.

**Reply by Lijesh-k** [reply]
Thank you @kumilingus
